File: mentorconnect/app/schemas/mentorship_request.py
"""
app/schemas/mentorship_request.py
-------------------------------------
Schemas for students sending mentorship requests and mentors accepting/rejecting them.
"""

from datetime import datetime
from typing import Optional
from pydantic import BaseModel, ConfigDict, Field

from app.models.enums import MentorshipRequestStatus


class MentorshipRequestCreate(BaseModel):
    mentor_id: int
    message: Optional[str] = Field(None, max_length=2000)


class MentorshipRequestRespond(BaseModel):
    """Mentor's response: accept or reject."""
    status: MentorshipRequestStatus = Field(..., description="Must be 'accepted' or 'rejected'")
    response_note: Optional[str] = Field(None, max_length=500)


class MentorshipRequestOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    student_id: int
    mentor_id: int
    message: Optional[str] = None
    status: MentorshipRequestStatus
    response_note: Optional[str] = None
    responded_at: Optional[datetime] = None
    created_at: datetime

File: mentorconnect-frontend/src/types/mentorshipRequest.ts
import { MentorshipRequestStatus } from './enums';
import { Mentor } from './mentor';
import { Student } from './student';

export interface MentorshipRequest {
  id: number;
  student: Student;
  mentor: Mentor;
  message?: string | null;
  status: MentorshipRequestStatus;
  responded_at?: string | null;
  response_note?: string | null;
  created_at: string;
}

export interface CreateMentorshipRequest {
  mentor_id: number;
  message?: string;
}

export interface RespondMentorshipRequest {
  status: MentorshipRequestStatus.ACCEPTED | MentorshipRequestStatus.REJECTED;
  response_note?: string;
}